packages/app/src/app/pages/common/Modals/PreferencesModal/Editor/VimModeSettings.tsx

```diff
@@ -0,0 +1,65 @@
+import { isSafari } from '@codesandbox/common/lib/utils/platform';
+import { Preference } from '@codesandbox/common/lib/components/Preference';
+import { Settings } from '@codesandbox/common/lib/components/Preview/types';
+import { Text } from '@codesandbox/components';
+import React, { useState } from 'react';
+
+import Modal from 'app/components/Modal';
+import { useAppState, useActions } from 'app/overmind';
+
+import { Alert } from '../../Common/Alert';
+
+import { PreferenceContainer } from '../elements';
+
+const SETTING_NAME: keyof Settings = 'vimMode';
+
+export const VimModeSettings: React.FC = () => {
+  const { settingChanged } = useActions().preferences;
+  const { settings } = useAppState().preferences;
+  const [showConfirmationModal, setShowConfirmationModal] = useState(false);
+
+  return (
+    <>
+      <PreferenceContainer disabled={isSafari}>
+        <Preference
+          title="Enable VIM extension"
+          type="boolean"
+          setValue={(value: boolean) => {
+            settingChanged({ name: SETTING_NAME, value });
+            setShowConfirmationModal(true);
+          }}
+          value={settings[SETTING_NAME]}
+        />
+
+        <Text
+          block
+          marginTop={2}
+          size={2}
+          style={{ maxWidth: '60%', lineHeight: 1.5 }}
+          variant="muted"
+        >
+          Use the command palette to control VIM
```

Comment:
I changed the text here a little because users don't need to know about the need to refresh when changing this setting until after they choose to change it.

The old text was:

"Toggling the VIM extension will require a refresh. When enabled, use the command palette to control VIM"